main.js

```diff
@@ -0,0 +1,212 @@
+	shuffleCards(cards);
+	//populate HTML
+	const fragment = document.createDocumentFragment();
+	for (let x = 0; x < num; x += 1) {
+		const cardContainer = document.createElement('div');
+		cardContainer.classList.add('card-container', difficulty);
+		const front = document.createElement('div');
+		const back = document.createElement('div');
+		front.classList.add('card', 'front');
+		back.classList.add('card', 'back');
+		const icon = document.createElement('span');
+		icon.classList.add('icon', 'material-symbols-outlined');
+		icon.setAttribute("name", cards[x]);
+		icon.innerText = cards[x];
+		back.appendChild(icon);
+		cardContainer.appendChild(front);
+		cardContainer.appendChild(back);
+		fragment.appendChild(cardContainer);
+	}
+	board.appendChild(fragment);
+	console.log(cards);
+}
+
+
+//-----CHOOSE DIFFICULTY
+function chooseDifficulty(){
+	[].forEach.call(difficulties, function(input){
+		input.nextElementSibling.classList.remove('checked');
+		console.log(input.nextElementSibling)
+		if (input.value === 'easy' && input.checked === true) {
+			NumberOfCards = 12;
+			difficulty = 'easy';
+			input.nextElementSibling.classList.add('checked');
+			board.classList.remove('normal-game', 'hard-game');
+			board.classList.add('easy-game');
+		} else if (input.value === 'normal' && input.checked === true) {
+			NumberOfCards = 24;
+			difficulty = 'normal';
+			input.nextElementSibling.classList.add('checked');
+			board.classList.remove('easy-game', 'hard-game');
+			board.classList.add('normal-game');
+		} else if (input.value === 'hard' && input.checked === true) {
+			NumberOfCards = 36;
+			difficulty = 'hard';
+			input.nextElementSibling.classList.add('checked');
+			board.classList.remove('easy-game', 'normal-game');
+			board.classList.add('hard-game');
+		}
+	});
+}
+
+function stopwatch(){
+	sec += 1;
+	let minutes = Math.floor(sec/60);
+	let seconds = sec % 60;
+	if (sec<10) {
+		timer.innerText = "00:0"+sec;
+	}
+	else if (sec<60) {
+		timer.innerText = "00:"+sec;
+	}
+	else if (sec<3600) {
+		timer.innerText = minutes+":"+seconds;
+	}
+}
+
+// -----CHECK WIN AND DISPLAY MODAL
+function checkwin(num) {
+	let won;
+	switch (difficulty) {
+		case 'easy' :
+			if (num === 6) {
+				won = true;
+			};
+			break;
+		case 'normal' :
+			if (num === 12) {
+				won = true;	
+			};
+			break;
+		case 'hard' :
+			if (num === 18){
+				won = true;
+			};
+			break;
+	};
+	if (won === true) {
+		setTimeout(() => {
+			//fill and show modal
+			document.getElementById('final-time').innerText = timer.innerText;
+			document.getElementById('final-moves').innerText = moves;
+			modal.classList.remove('hide');
+			//stop the stopwatch
+			clearInterval(setTimer);
+		}, 500);
+	}
+}
+
+
+//-----CORE OF THE GAME - CHECKING CARDS AND COUNTING MOVES
+function matchChecker(event){
+	//making sure the clicked target is a card and prevent doubleclicking
+	if (event.target.classList.contains('card') && !event.target.classList.contains('front-open')) {
```

Comment:
Aș inversa `if`-ul, pentru ca logica principală a unei funcții să nu fie într-un `if`.

```suggestion
	if (!event.target.classList.contains('card') || event.target.classList.contains('front-open')) {
	  return;
	}
```